src/components/ReconnectPrompt.svelte

```diff
@@ -5,74 +5,44 @@
  -->
 
 <script lang="ts">
-  import { horizontalSlide } from '../script/transitions';
   import StandardButton from '../components/StandardButton.svelte';
   import { state } from '../script/stores/uiStore';
   import { t } from '../i18n';
-  import { btPatternInput, btPatternOutput } from '../script/stores/connectionStore';
-  import MBSpecs from '../script/microbit-interfacing/MBSpecs';
-  import Microbits from '../script/microbit-interfacing/Microbits';
   import { DeviceRequestStates } from '../script/stores/connectDialogStore';
-
-  let reconnectText: string;
-  let reconnectButtonText: string;
-
-  state.subscribe(s => {
-    if (s.reconnectState === DeviceRequestStates.INPUT) {
-      reconnectText = $t('disconnectedWarning.input');
-      reconnectButtonText = $t('disconnectedWarning.reconnectButton.input');
-    } else if (s.reconnectState === DeviceRequestStates.OUTPUT) {
-      reconnectText = $t('disconnectedWarning.output');
-      reconnectButtonText = $t('disconnectedWarning.reconnectButton.output');
-    }
-  });
-  // When disconnected by lost connection, offer the option to attempt to reconnect
-  let hideReconnectMessageAfterTimeout = false;
-  state.subscribe(s => {
-    if (s.offerReconnect) {
-      hideReconnectMessageAfterTimeout = true;
-    }
-  });
-
-  const reconnect = (connectState: DeviceRequestStates) => {
-    hideReconnectMessageAfterTimeout = false;
-    console.assert(connectState != DeviceRequestStates.NONE);
-    const pairingPattern =
-      connectState === DeviceRequestStates.INPUT ? $btPatternInput : $btPatternOutput;
-    const name = MBSpecs.Utility.patternToName(pairingPattern);
-
-    const connect = () => {
-      if (connectState == DeviceRequestStates.INPUT) {
-        return Microbits.assignBluetoothInput(name);
-      }
-      return Microbits.assignOutput(name);
-    };
-
-    void connect().then(didSucceed => {
-      if (didSucceed) {
-        $state.offerReconnect = false;
-      }
-    });
+  import StandardDialog from './dialogs/StandardDialog.svelte';
+  import { startConnectionProcess } from '../script/stores/connectDialogStore';
+
+  export let isOpen: boolean = false;
+
+  const dialogText =
```

Comment:
This needs to be reactive, otherwise it always says "Reconnect output".
```suggestion
$: dialogText =
```